qa/L0_java_memory_growth/Simple.java

```diff
@@ -0,0 +1,847 @@
+              irequest, input0, datatype, input0_shape, input0_shape.length),
+          "setting input 0 meta-data for the request");
+      FAIL_IF_ERR(
+          TRITONSERVER_InferenceRequestAddInput(
+              irequest, input1, datatype, input1_shape, input1_shape.length),
+          "setting input 1 meta-data for the request");
+
+      String output0 = is_torch_model[0] ? "OUTPUT__0" : "OUTPUT0";
+      String output1 = is_torch_model[0] ? "OUTPUT__1" : "OUTPUT1";
+
+      FAIL_IF_ERR(
+          TRITONSERVER_InferenceRequestAddRequestedOutput(irequest, output0),
+          "requesting output 0 for the request");
+      FAIL_IF_ERR(
+          TRITONSERVER_InferenceRequestAddRequestedOutput(irequest, output1),
+          "requesting output 1 for the request");
+
+      // Create the data for the two input tensors. Initialize the first
+      // to unique values and the second to all ones.
+      BytePointer input0_data;
+      BytePointer input1_data;
+      if (is_int[0]) {
+        IntPointer[] p0 = {null}, p1 = {null};
+        GenerateInputData(p0, p1);
+        input0_data = p0[0].getPointer(BytePointer.class);
+        input1_data = p1[0].getPointer(BytePointer.class);
+      } else {
+        FloatPointer[] p0 = {null}, p1 = {null};
+        GenerateInputData(p0, p1);
+        input0_data = p0[0].getPointer(BytePointer.class);
+        input1_data = p1[0].getPointer(BytePointer.class);
+      }
+
+      long input0_size = input0_data.limit();
+      long input1_size = input1_data.limit();
+
+      Pointer input0_base = input0_data;
+      Pointer input1_base = input1_data;
+
+      FAIL_IF_ERR(
+          TRITONSERVER_InferenceRequestAppendInputData(
+              irequest, input0, input0_base, input0_size, requested_memory_type,
+              0 /* memory_type_id */),
+          "assigning INPUT0 data");
+      FAIL_IF_ERR(
+          TRITONSERVER_InferenceRequestAppendInputData(
+              irequest, input1, input1_base, input1_size, requested_memory_type,
+              0 /* memory_type_id */),
+          "assigning INPUT1 data");
+
+      // Perform inference...
+      {
+        CompletableFuture<TRITONSERVER_InferenceResponse> completed = new CompletableFuture<>();
+        futures.put(irequest, completed);
+
+        FAIL_IF_ERR(
+            TRITONSERVER_InferenceRequestSetResponseCallback(
+                irequest, allocator, null /* response_allocator_userp */,
+                inferResponseComplete, irequest),
+            "setting response callback");
+
+        FAIL_IF_ERR(
+            TRITONSERVER_ServerInferAsync(
+                server, irequest, null /* trace */),
+            "running inference");
+
+        // Wait for the inference to complete.
+        TRITONSERVER_InferenceResponse completed_response = completed.get();
+        futures.remove(irequest);
+
+        FAIL_IF_ERR(
+            TRITONSERVER_InferenceResponseError(completed_response),
+            "response status");
+
+        Check(
```

Comment:
We should probably make Check only run based on a command line boolean. That way we can break up the test into a sanity test to make sure that the full Simple example has no memory leak (run for a small number of iterations) and a high-iteration memory leak test just for the inference. Right now, I suspect Check is likely taking up a lot of the time per inference, which I suspect we don't want. What do you think?